ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Ledger/Stake.hs

```diff
@@ -78,14 +80,17 @@ totalStakes addrDist = foldl f Map.empty
 
 -- | Stake distribution where every address has equal state
 equalStakeDist :: AddrDist -> StakeDist
-equalStakeDist = StakeDist
-               . Map.fromList
-               . mapMaybe (nodeStake . snd)
-               . Map.toList
+equalStakeDist ad =
+    StakeDist $
+    Map.fromList $
+    mapMaybe (nodeStake . snd) $
+    Map.toList ad
   where
     nodeStake :: NodeId -> Maybe (CoreNodeId, Rational)
     nodeStake (RelayId _) = Nothing
-    nodeStake (CoreId i)  = Just (i, 1)
+    nodeStake (CoreId i)  = Just (i, recip (fromIntegral n))
+
+    n = sum $ map (length . nodeStake) $ Map.elems ad
```

Comment:
Is `sum . map length` instead of `length . filter isJust`  too cheeky?

Reply:
I think so. `length . filter isJust` is compiled validated documentation.